src/ODESolverTypes.f90

```diff
@@ -624,6 +642,12 @@ SUBROUTINE solve_bdf(f,myLS,ord,nstep,t,dt,yf,ydot,tol,ist,bdf_hist,updateJ_in,k
     keepSolving=.FALSE.
   ENDIF
```

Comment:
I think we could eliminate keepSolving.  That would result in an erroneous result since the idea was to allow the ODE integration to continue even if the implicit solve failed.  maxIter is required to prevent the implicit solve step from getting stuck in an inf loop.